keyboards/shorty/keymaps/default/keymap.c

```diff
@@ -0,0 +1,18 @@
+// Copyright 2023 QMK
+// SPDX-License-Identifier: GPL-2.0-or-later
+#include QMK_KEYBOARD_H
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [0] =   { ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(KC_MS_WH_DOWN,KC_MS_WH_UP)  }
+    //                  Encoder 1                                     Encoder 2
+};
+#endif
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
+        KC_P7,   KC_P8,   KC_P9,    
+        KC_P4,   KC_P5,   KC_P6,     
+        KC_P1,   KC_P2,   KC_P3,  
+        KC_A,    KC_NO,   KC_NO, // encoder button1
+        KC_D,    KC_NO,   KC_NO  // encoder button2
```

Comment:
Use of `KC_NO` within the only layer? This would suggest your layout macro is wrong and is not correctly abstracting the physical layout.

Reply:
The encoder click pins are not connected to the main matrix. Instead, they use their own pins, `GP29` and `GP2`. To integrate them into the matrix, we've used a null value for their corresponding `col` and emulated a 5x5 matrix based on the actual PCB layout. While this non-standard PCB layout isn't ideal, given our significant investment in over 100 printed units, this workaround proved to be the most practical solution so far.

Reply:
> and emulated a 5x5 matrix based on the actual PCB layout

But nothing is mapped to those `null` matrix locations.

```json
                {"matrix": [3, 0], "x": 0, "y": 3},
...
                {"matrix": [4, 0], "x": 0, "y": 4},
```

These matrix locations are `[row, col]`. These are would have to be  `3,3` / `3,4` and `4,3` / `4,4` for your logic to stand up. Your config isnt doing what you think it is....

Reply:
I think I'll try to modify the code to understand what I did and why. Thanks for your time and assistance. Really appreciated.

Reply:
Hi @zvecr, I've made the changes you suggested. You were right, the previous way was unnecessary. I've also modified the layout. Please let me know if you think there's still room for improvement.